Add frontend component library ADR (#614)
azavea/echo-locator: doc/arch/adr-002-frontend-component-library.md

# Frontend Component Library

## Context

*Note: Below copied with small edits from the initial component library evaluation, [found here in the Issue 603 discussion.](https://github.com/azavea/echo-locator/issues/603)*

We're rewriting the frontend [with React and vite](https://github.com/azavea/echo-locator/issues/599#issuecomment-2736833829) and should bring in a component library to ensure consistency across the frontend, speed up development, and handle things like accessibility and styling out-of-the-box. This is a mobile-first rewrite and users may be on slower network speeds or older devices as well as may be using assistive technologies. Knowing this the follow are important considerations:

- Accessibility
- Mobile-First and Responsiveness
- Internationalization (Spanish and Chinese translations in frontend)
- Library maintainability/stability
- Official figma kit (to bridge designs and development)

> Also an important considerations is the ability to have other developers jump on quickly, so the chosen library should have a well-documented library or strong team familiarity and examples.

Reviewer: ```suggestion
Another important consideration is the ability to have other developers jump on quickly, so the chosen library should have a well-documented library or strong team familiarity and examples.
```

> We should move forward with using Tailwind CSS for this frontend. We have growing familiarity with Tailwind and using a utility-first styling approach can help speed development and maintain consistency across the application, especially if taking advantage of its theme variables. Tailwind is also mobile-first and automatically removes all unused CSS when building for production so our final CSS bundle can be smaller.

Reviewer: ```suggestion
We should move forward with using [Tailwind CSS](https://tailwindcss.com) for this frontend. We have growing familiarity with Tailwind and using a utility-first styling approach can help speed development and maintain consistency across the application, especially if taking advantage of its [theme variables](https://tailwindcss.com/docs/theme). Tailwind is also mobile-first and automatically removes all unused CSS when building for production so our final CSS bundle can be smaller.
```
